nep-22.mediawiki

```diff
@@ -0,0 +1,73 @@
+<pre>
+  NEP: 22
+  Title: Contract Update Function
+  Author: Owen Zhang <zhangzhihong@ngd.neo.org>, Fernando Díaz Toledano <shargon@gmail.com>, Erik Zhang <erik@neo.org>, Roman Khimov <roman@nspcc.ru>
+  Type: Standard
+  Status: Accepted
+  Created: 2021-04-19
+</pre>
+
+
+==Abstract==
+
+This proposal outlines a standard for the NEO blockchain that will provide systems with a generalized interaction mechanism for smart contract update.
+
+==Motivation==
+
+As the Neo blockchain scales, Smart Contract Update will become increasingly important. We need a standard for <code>update</code> method in contract to callback native <code>ContractManagement</code> correctly.
+
+==Specification==
+
+All contracts SHOULD implement this NEP in case of any bug hotfix or improvement.
```

Comment:
Maybe explain current consequences if not implemented in the first version.

Reply:
Done

Reply:
It can be implemented in a different way, I mean non-NEP-22 method that still does the update. Also, the way it's worded now is like "if you have a bug, please implement this method", while in fact it's too late if you have a bug and no way to update the contract.